clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -178,8 +186,22 @@ public void close()  {
 
     private TransportLayer buildTransportLayer(String id, SelectionKey key, SocketChannel socketChannel) throws IOException {
         if (this.securityProtocol == SecurityProtocol.SASL_SSL) {
-            return SslTransportLayer.create(id, key,
-                sslFactory.createSslEngine(socketChannel.socket().getInetAddress().getHostName(), socketChannel.socket().getPort()));
+            long before = System.nanoTime();
+            String hostname = socketChannel.socket().getInetAddress().getHostName();
+            long after = System.nanoTime();
+            long totalMs = (after - before) / 1_000_000;
+            if (totalMs > 0) {
+                requestLogger.debug(String.format("%s took %d milliseconds", totalMs));
+            }
+
+            before = System.nanoTime();
```

Comment:
Could we just reuse "after" obtained in line 191?

Reply:
Yes, thanks!